FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/Services/AuthService.swift

```diff
@@ -96,6 +92,40 @@ public final class AuthService {
   private var listenerManager: AuthListenerManager?
   private var signedInCredential: AuthCredential?
 
+  private var providers: [any ExternalAuthProvider] = []
+  public func register(provider: any ExternalAuthProvider) {
+    providers.append(provider)
+  }
+
+  public func renderButtons(spacing: CGFloat = 16) -> AnyView {
+    AnyView(
+      VStack(spacing: spacing) {
+        ForEach(providers, id: \.id) { provider in
+          provider.authButton()
+        }
+      }
+    )
+  }
+
+  @ViewBuilder
+  public var googleButton: some View {
+    if googleProvider != nil {
+      // For purpose of demonstration
+      // This produces "Type 'any View' cannot conform to 'View'"
```

Comment:
@peterfriese - the problem is when we try to render from core package, Xcode produces this compiler error. I don't know how to get around it.

Reply:
Can you take a look at https://gist.github.com/peterfriese/d9745f366fb4f857daf077e93b1eb01f and see if this approach helps?

Reply:
@peterfriese - here is the commit attempting to follow your example: [`8c38462` (#1250)](https://github.com/firebase/FirebaseUI-iOS/pull/1250/commits/8c3846276c21769bbe19530eced2824f8f37f97c)